src/odemis/driver/actuator.py

```diff
@@ -1366,3 +1366,510 @@ def terminate(self):
+
+        # If the axis can be referenced => do it now (and move to a known position)
+        # In case of cyclic move always reference
+        if not self._referenced.get(axis, True) or (self._cycle and axis in self._referenced):
+            # The initialisation will not fail if the referencing fails
+            f = self.reference({axis})
+            f.add_done_callback(self._on_referenced)
+
+    def updateMetadata(self, md):
+        for key, value in md.items():
+            if isinstance(value, float) and (0.0 <= abs(value) <= self._cycle/2):
+                super(RotationActuator, self).updateMetadata(md)
+                self._updatePosition()
+                logging.debug("reporting metadata entry %s with value %s." % (value, key))
+            else:
+                logging.exception("value %s for metadata entry %s is not allowed. "
+                                  "Value should be in range -%s/2 and +%s/2." % (value, key, self._cycle, self._cycle))
+                raise ValueError("value %s for metadata entry %s is not allowed." % (value, key))
+
+    def _on_referenced(self, future):
+        try:
+            future.result()
+        except Exception as e:
+            self._child.stop({self._caxis})  # prevent any move queued
+            self.state._set_value(e, force_write=True)
+            logging.exception(e)
+
+    def _updatePosition(self, _=None):
+        """
+        update the position VA
+        """
+        p = self._child.position.value[self._caxis] - self._metadata[model.MD_POS_COR]
+        if self._cycle is not None:
+            p %= self._cycle
+        self._position[self._axis] = p
+        real_pos = self._position[self._axis]
+        pos = {self._axis: real_pos}
+        logging.debug("reporting position %s", pos)
+        self.position._set_value(pos, force_write=True)
+
+    def _updateReferenced(self, _=None):
+        """
+        update the referenced VA
+        """
+        self._referenced[self._axis] = self._child.referenced.value[self._caxis]
+        # .referenced is copied to detect changes to it on next update
+        self.referenced._set_value(self._referenced.copy(), force_write=True)
+
+    @isasync
+    def moveRel(self, shift):
+        if not shift:
+            return model.InstantaneousFuture()
+        self._checkMoveRel(shift)
+        raise NotImplementedError("Relative move on fixed positions axis not supported")
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        Move the actuator to the defined position in m for each axis given.
+        pos dict(string-> float): name of the axis and position in m
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        self._checkMoveAbs(pos)
+        pos = self._applyInversion(pos)
+        f = CancellableFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, pos)
+        f.task_canceller = self._cancelMovement
+
+        return f
+
+    def _cancelMovement(self, future):
```

Comment:
Cancelling is hard. Here, if cancel() is called at the beginning of the move (for instance, while it's doing the referencing), it will report that it got cancelled too late... and then continue on the move.
=> Let's keep it simple for now and not support cancelling (and you can drop `self._child_future` at the same time).

Reply:
should I maybe keep the code so far by using comments and pass statement when the function is called?